plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2731,13 +2890,26 @@ public Optional<ConstraintApplicationResult<ConnectorTableHandle>> applyFilter(C
             return Optional.empty();
         }
 
+        replaceTableHandle(tableHandle, newHandle);
         return Optional.of(new ConstraintApplicationResult<>(
                 newHandle,
                 newUnenforcedConstraint.transformKeys(ColumnHandle.class::cast),
                 extractionResult.remainingExpression(),
                 false));
     }
 
+    private void replaceTableHandle(DeltaLakeTableHandle tableHandle, DeltaLakeTableHandle newHandle)
```

Comment:
What happens if the concurrent operations access this method at the same time? I suppose it will cause ConcurrentModificationException. `synchronized` looks missing?